src/haxeLanguageServer/features/haxe/codeAction/TokenTreeUtils.hx

```diff
@@ -16,6 +20,13 @@ class TokenTreeUtils {
 		return fun.tok.match(Kwd(KwdFunction));
 	}
 
+	public static function isCallPOpen(pOpen:TokenTree):Bool {
+		if (pOpen.tok != POpen)
+			return false;
+		final name = pOpen.parent ?? return false;
+		return name.tok.match(Const(CIdent(_)));
+	}
```

Comment:
there is a `TokenTreeCheckUtils.getPOpenType()` function in tokentree lib, that returns an enum value telling you what type of `POpen` you have. it doesn't really check if the token passed is a `POpen`, because it assumes that the caller only calls it on `POpen` tokens.
I'm not sure if that might give better results.

there is also `getBrOpenType` but that doesn't give you info about what kind of block a `BrOpen` token  represents.